### ISSUE TITLE
Doesn't work for children prop

### ISSUE THREAD

**Issue by ramoreyiw** [getcouped/eslint-plugin-react-google-translate#1]
Thanks for the great work, really useful.

I've noticed an edge case that seems quite common in my code.

It relates to when you pass a string as a children prop to a component that renders next to a sibling.

For example:

`<Button>Children as string</Button>`

```
    <Button>
      {loading && <Spinner />}
      {children}
    </Button>
```

This would cause an error with google translate, but does not show an error with the linting.

I don't have any immediate thoughts on a solution but I thought it might be nice to start an issue on it.

**Comment by andreivictor** [comment]
Same issue when using a translate function, eg:

```
<Button>
  {loading && <Spinner />}
  { t('translation-key') }
</Button>
 ```

**Comment by alistair-coup** [comment]
Thanks for the comments. I will address the 2 comments separately as the answers are different

@ramoreyiw with regards to the `children` prop, it is difficult with static analysis to say with confidence what the run time value of a property will be. Currently we can identify that we have a variable called `children`, though we don't know what type the value is. The issue we are presented with here is that within React applications, the value of `children` is invariably a `ReactNode` and so simply reporting it's presence beside a conditionally rendered node would often provide the user with a false positive, something we wish to minimise.

I see 2 steps forward here with this. The first is to update the `README.md` to include this in the list of known limitations, which I will do just now. The second would be to introduce the TypeScript ESLint parser to the ESLint rule codebase, which would give us the superpower of being able to identify values by their type. Given the nature of the ESLint rule I see this as an improvement and have taken a note of the potential change. PRs are welcome though, as this is low to medium priority for us.

@andreivictor thanks for the feedback. Accommodating i18n has been on our radar for some time now. Since we can confidently say that the return value of the `t` function is a string, and due to it's popularity, I think it makes sense to include this in the rule. I will take care of this just now.

I'm going to close the issue for now as I consider it resolved.
